packages/opentelemetry-exporter-collector/src/platform/node/types.ts

```diff
@@ -24,5 +24,11 @@ import { CollectorExporterConfigBase } from '../../types';
 export interface CollectorExporterNodeConfigBase
   extends CollectorExporterConfigBase {
   keepAlive?: boolean;
+  compression?: CompressionAlgorithm;
   httpAgentOptions?: http.AgentOptions | https.AgentOptions;
 }
+
+export enum CompressionAlgorithm {
```

Comment:
General comment, why define the enum values as strings?
Why not just a numeric value as the "NAME" will still be available is the resulting code and if required we can always look it up in config via  uppercasing ... Just a thought.

Reply:
Generally i advise using string to avoid problem when trying to update available value (that can share the same number even if values are different across versions)

Reply:
as far as I know normal enums are hard to use for non typescript users.

Reply:
@Flarna I know that's true for `const enum`s but I'm not sure it's the case for regular enums. Do you have a particular case you're thinking of?

Reply:
I think I know what you mean after thinking a little more. You're not thinking they can't do `CompressionAlgorithm.GZIP` but that `1` doesn't mean anything to them and passing a string `'gzip'` is more clear?

Reply:
Normal enums are exported as objects so a JS user can use `CompressionAlgorithm.GZIP`. A JS user can also use the underlying constant (e.g. `gzip`).
The main difference is documentation. A typescript user can't to much wrong as editor will tell the allowed values to use and compiler checks again.

For JS users it's needed to document that an object named `CompressionAlgorithm` is exported incl. the properties (`GZIP` and `NONE`) referring to the constants to use.

This works fine but often JS APIs document the actual values to use instead referring to an object holding the constants (would be `gzip`/`none` here).

Standard const enums are really bad as JS users have to use the numeric values then.
const enums with strings as values would require JS users to use these strings in their code.